config/src/main/java/com/quorum/tessera/config/adapters/EncryptedStringAdapter.java

```diff
@@ -0,0 +1,43 @@
+package com.quorum.tessera.config.adapters;
+
+import com.quorum.tessera.config.util.ConfigSecretReader;
+import org.jasypt.encryption.pbe.PBEStringEncryptor;
+import org.jasypt.encryption.pbe.StandardPBEStringEncryptor;
+import org.jasypt.properties.PropertyValueEncryptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.bind.annotation.adapters.XmlAdapter;
+
+public class EncryptedStringAdapter extends XmlAdapter<String, String> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EncryptedStringAdapter.class);
+
+    private final PBEStringEncryptor encryptor = new StandardPBEStringEncryptor();
+
+    @Override
+    public String unmarshal(String textToDecrypt) {
+
+        if (PropertyValueEncryptionUtils.isEncryptedValue(textToDecrypt)) {
+
+            if (!((StandardPBEStringEncryptor) encryptor).isInitialized()) {
```

Comment:
Should we not have the implementation as the member variable? No point casting every time for the sake of having an interface as the member variable.
I guess if you need convincing, then we do depend on this particular implementation, so that is what we should include, we don't depend on just the interface.